src/Parser/Core/Modules/DamageMitigation/DamageMitigation.js

```diff
@@ -0,0 +1,380 @@
+    debug && console.log('Actual armor: ' + event.armor + ', armorTracker armor: ' + this.armor.armor);
+    
+    // Add passives.
+    let mitigations = this.passives.slice();
+
+    // Check for active buffs.
+    mitigations = mitigations.concat(this.getBuffs(event));
+
+    // Check for active debuffs.
+    mitigations = mitigations.concat(this.getDebuffs(event));
+
+    // Filter by AOE or non-AOE.
+    mitigations = this.filterAOE(event, mitigations);
+
+    // Filter by spell school.
+    mitigations = this.filterSchool(event, mitigations);
+
+    if (debug) {
+      let names = '';
+      mitigations.forEach(e => {
+        names += e.name + ', ';
+      });
+      console.log('Reductions: ' + names);
+    }
+
+    // No active mitigations. Should never happen since vers is always included.
+    if (mitigations.length === 0) {
+      console.error('No mitigations found. This should never happen since versatility should always be included (even when you have 0).');
+      return;
+    }
+
+    const additiveReducer = (accumulator, reduction) => accumulator + this.getMitigation(reduction, event);
+    const multiplicativeReducer = (accumulator, reduction) => accumulator * (1 - this.getMitigation(reduction, event));
+
+    let percentMitigated = mitigated/event.unmitigatedAmount;
+    let multiplicative = 1 - mitigations.reduce(multiplicativeReducer, 1);
+    
+    // The remaining mitigation not accounted for. (can be also be negative!)
+    let unknown = 1 - (1 - percentMitigated) / (1 - multiplicative);
+
+    // If the actual percent mitigated is less than expected, it is likely a physical ability not being reduced by armor.
+    if (unknown < - BUFFER_PERCENT) {
+      // If it is physical damage, try to match without armor instead.
+      if (event.ability.type === 1 && event.ability.name !== 'Melee') {
+        mitigations = mitigations.filter(e => e.id !== -1001);
+        // Recalculate
+        multiplicative = 1 - mitigations.reduce(multiplicativeReducer, 1);
+        unknown = 1 - (1 - percentMitigated) / (1 - multiplicative);
+      }
+      if (unknown < - BUFFER_PERCENT) {
+        debug && console.warn('The actual percent mitigated was lower than expected given the mitigations active at the time. Actual: ' + formatPercentage(percentMitigated) + '%, Expected: ' + formatPercentage(multiplicative) + '%');
+      }
+    }
+
+    // If the actual percent mitigated is higher than expected, try to assign the unknown mitigation.
+    if (unknown > BUFFER_PERCENT) {
+      // If the player is using a shield and the hit was fully absorbed, assume the remaining mitigation was block (Block amount is not in the log when the hit fully absorbed).
+      if (event.amount === 0 && (this.selectedCombatant.specId === SPECS.PROTECTION_PALADIN.id || 
```

Comment:
I'd prefer it if this file didn't include effects that are class/spec specific. I believe core shouldn't know about class/spec specific things unless they can be applied to others (e.g. externals).